### ISSUE TITLE
Psalm emits InaccessibleProperty and RedundantPropertyInitializationCheck for checks and mutating code in the "__wakeup"

### ISSUE THREAD

**Issue by andrew-demb** [vimeo/psalm#10182]
### Discussed in https://github.com/vimeo/psalm/discussions/10180

<div type='discussions-op-text'>

<sup>Originally posted by **andrew-demb** September  5, 2023</sup>
https://psalm.dev/r/0b9d93cacf

```php
<?php

/**
 * @psalm-immutable
 */
class Some
{
    // it was a `?string $locale` in past
    private string $locale;
    
    public function __construct(
        string $locale,
    ) {
        $this->locale = $locale;
    }

    public function __wakeup(): void
    {
    	// it was a `?string $locale` in past
        if (false === isset($this->locale)) {
            $this->locale = '';
        }
    }
}
```

```
Psalm output (using commit [37cc4fd](https://github.com/vimeo/psalm/commit/37cc4fd)): 

ERROR: [RedundantPropertyInitializationCheck](https://psalm.dev/261) - 20:13 - Property $this->locale with type string should already be set in the constructor

ERROR: [InaccessibleProperty](https://psalm.dev/054) - 21:13 - Some::$locale is marked readonly
```

What's the proper way to handle it? 
Suppress both issues or psalm may allow such a checks and updates for magic method like a `__wakeup`?</div>

As of https://github.com/vimeo/psalm/discussions/10180#discussioncomment-6912965 - the current issue requests for a feature in Psalm - don't emit such errors for code like above in the `__wakeup`.
